### ISSUE TITLE
A handouts feature similar to how Maps work.

### ISSUE THREAD

**Issue by adventuremagic123** [RPTools/maptool#2045]
**Is your feature request related to a problem? Please describe.**
As a GM, I'm tempted to use the maps feature for handouts -- but this is awkward and not exactly what I want.

**Describe the solution you'd like**
 I would like to have handouts stored as HTML text or an image or both and have them selectable like maps.  When I select a handout, it should be displayed to the players on top of whatever map is displayed or, if easier, instead of the current map.  Then, I should have a button, perhaps, to dismiss the handout -- in which case the previous map is made visible again.  The handouts selection control should be next to the map selection control.

**Describe alternatives you've considered**
Using the map feature to display handouts is an alternative I've considered, but then adding handouts there would make the maps harder to use.  I tried implementing macros using an image table, but that has been very awkward and has fallen out of use.

**Additional context**
Most VTTs have a handouts feature, so I strongly recommend a handouts feature for MapTool.

**Comment by Azhrei** [comment]
Don't frames currently handle this?  They can be minimized, docked, or stacked.  However, they don't appear when in full screen mode.

It also seems like overlays could accomplish all of this.  If you want the content to be dynamic, particularly if you want to integrate some type of form or user input, they can be a little difficult to use right now.  You could even use the z-order supported in overlays to have stacks of handouts and switch between them.

I suppose a Few pages on the wiki to show samples of how to set this up would helpful...

**Comment by adventuremagic123** [comment]
Yes, I think that would work.  It would be nicer for the non-programmers to have a handouts feature without writing macros, since handouts are probably base-level functionality in the same way as maps.

For example, RPG players would use the feature and that's probably obvious.  But, since MapTool's audience target is beyond RPG, people implementing other things like board games could use handouts for game instructions.

**Comment by adventuremagic123** [comment]
@Azhrei Actually, it's been so long since I've used it, I forgot what I had done.  I implemented a handouts management system that uses an image table and frames.  It can assign or unassign handouts to all players or individual players.  I used an image table because I wasn't sure how else to give my scripts access to images.  The GM can drop a handout in the image table and then assign it to players.

The problem with using the image table was that I kept losing images at the bottom of it at random points in the furure when I'd reload the campaign.  Therefore, my handouts feature would suddenly become broken even though it was working perfectly -- and I hadn't touched anything since it was working.  I think in retrospect it had to do with my not realizing or remembering (more likely) that image tables are stored in the campaign settings.  I have a development campaign.  When I'm happy with my latest change there, I export campaign settings to the campaign where I run my game.  So, I kept smashing my image table.

Given all that, I need to ammend my feature request.

I think a handouts feature is very important for MapTool for the reasons I stated above.  It's a base feature everyone can use.  It also allows GMs to put handouts in front of their players -- which I don't think is possible with macros and frames (where players would have to invoke the macro to see the handout).

**Comment by Akaku-Keninari** [comment]
> I think a handouts feature is very important for MapTool for the reasons I stated above. It's a base feature everyone can use. It also allows GMs to put handouts in front of their players just as they do with maps -- which I don't think is possible with macros and frames (where players would have to invoke the macro to see the handout).

execLink() can open a frame for everyone, can't it? It can open an overlay at least.

**Comment by cwisniew** [comment]
> > I think a handouts feature is very important for MapTool for the reasons I stated above. It's a base feature everyone can use. It also allows GMs to put handouts in front of their players just as they do with maps -- which I don't think is possible with macros and frames (where players would have to invoke the macro to see the handout).
> 
> execLink() can open a frame for everyone, can't it? It can open an overlay at least.

Yes but it's nice to have a handout functionality for those that don't want to get into macros and scripting.
